routes/user.js

```diff
@@ -41,312 +41,280 @@ var config = require('config')
-                }
-            });
-            var mailOptions = {
-                to: user.email,
-                from: 'noreply@theotech.org',
-                subject: 'Reset Password',
-                text: 'You are receiving this because you (or someone else) have requested the reset of the password for your account.\n\n' +
-                    'Please click on the following link, or paste this into your browser to complete the process:\n\n' +
-                    'http://' + req.headers.host + '/user/reset/' + token + '\n\n' +
-                    'If you did not request this, please ignore this email and your password will remain unchanged.\n'
-            };
-            smtpTransport.sendMail(mailOptions, function(err) {
-                req.flash('info', 'An e-mail has been sent to ' + user.email + ' with further instructions.');
-                done(err, 'done');
-            });
-        }
-    ], function(err) {
-        if (err) return next(err);
-        res.redirect('/user/forgot');
-    });
-});
-
-
-router.get('/reset/:token', function(req, res, next) {
-    User.findOne({
-        resetPasswordToken: req.params.token,
-        resetPasswordExpires: {
-            $gt: Date.now()
-        }
-    }, function(err, user) {
-        if (!user) {
-            req.flash('error', 'Password reset token is invalid or has expired.');
-            return res.redirect('/user/forgot');
-        }
-        res.render('reset', {
-            user: req.user
+            },
+            function(token, done) {
+                User.findOne({
+                    email: req.body.email
+                }, function(err, user) {
+                    if (!user) {
+                        req.flash('error', 'No account with that email address exists.');
+                        return res.redirect('/user/forgot');
+                    }
+
+                    user.resetPasswordToken = token;
+                    user.resetPasswordExpires = Date.now() + 3600000; // 1 hour
+
+                    user.save(function(err) {
+                        done(err, token, user);
+                    });
+                });
+            },
+            function(token, user, done) {
+                var smtpTransport = nodemailer.createTransport('SMTP', {
+                    service: 'SendGrid',
+                    auth: {
+                        user: process.env.SENDGRID_USER || config.get('emailVerification.user'),
+                        pass: process.env.SENDGRID_PASS || config.get('emailVerification.pass')
+                    }
+                });
+                var mailOptions = {
+                    to: user.email,
+                    from: 'noreply@theotech.org',
+                    subject: 'Reset Password',
+                    text: 'You are receiving this because you (or someone else) have requested the reset of the password for your account.\n\n' +
```

Comment:
Change the messaging: 

Someone has requested to reset the password for your account at glorify.cc

If you did not request this, you can ignore this message. 
Otherwise, please follow this link to reset your password:
'https://' + req.headers.host + '/user/reset/' + token + ...

Notice I'm using https. If you can detect prod vs dev NODE_ENV, then you can use http in dev and https in prod.